stdlib/src/collections/list.mojo

```diff
@@ -197,6 +197,56 @@ struct List[T: CollectionElement](
     # Operator dunders
     # ===-------------------------------------------------------------------===#
 
+    fn __eq__[T2: ComparableCollectionElement](self: List[T], other: List[T2]) -> Bool:
```

Comment:
`ComparableCollectionElement` seems to require more than just `__eq__` and `__ne__` but it is the existing composition with `CollectionElement` hence I am using it.

Reply:
Now using `EqualityComparableCollectionElement`.